test/integration/PaymentTypes/InvoiceGuaranteedTest.php

```diff
@@ -0,0 +1,95 @@
+<?php
+/** @noinspection PhpUnhandledExceptionInspection */
+/** @noinspection PhpDocMissingThrowsInspection */
+/**
+ * This class defines integration tests to verify interface and functionality of the payment method Invoice Secured.
+ *
+ * Copyright (C) 2020 - today Unzer E-Com GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * @link  https://docs.unzer.com/
+ *
+ * @author  Simon Gabriel <development@unzer.com>
+ *
+ * @package  UnzerSDK\test\integration\PaymentTypes
+ */
+namespace UnzerSDK\test\integration\PaymentTypes;
+
+use UnzerSDK\Exceptions\UnzerApiException;
+use UnzerSDK\Resources\PaymentTypes\InvoiceSecured;
+use UnzerSDK\Resources\TransactionTypes\Charge;
+use UnzerSDK\test\BaseIntegrationTest;
+
+class InvoiceGuaranteedTest extends BaseIntegrationTest
+{
+    /**
+     * Verify, backwards compatibility regarding fetching payment type and map it to invoice secured class.
+     *
+     * @test
+     */
+    public function ivgTypeShouldBeFechable(): InvoiceSecured
+    {
+        $ivgMock = $this->getMockBuilder(InvoiceSecured::class)->setMethods(['getUri'])->getMock();
```

Comment:
Add to changelog

Reply:
I dont realy saw this noetworthy in the changelog. I would prefere to let it go under "several minior improvements".

Reply:
Will be added with the next part of the UPL integration